index.js

```diff
@@ -141,14 +144,21 @@ precinct.paperwork = function(filename, options) {
   var ext = path.extname(filename);
   var type;
 
-  if (ext === '.css' || ext === '.scss' || ext === '.sass' || ext === '.less' || ext === '.ts') {
+  if (ext === '.css' || ext === '.scss' || ext === '.sass' || ext === '.less' || ext === '.ts' || ext === '.tsx') {
     type = ext.replace('.', '');
-
   } else if (ext === '.styl') {
     type = 'stylus';
   }
 
   options.type = type;
+  
+  if (type === 'tsx') {
```

Comment:
Precinct tries to remain very configurable and prescriptive. See the usage docs: https://github.com/pahen/madge#configuration for examples.

As such, we should document the possibility of setting `tsx.ecmaFeatures.jsx = true` in the Readme, but should avoid setting it on (at least within precinct) by default. 

How are you using/consuming precinct? Are you using some module that uses precinct? If so, consider setting the option at that consumer layer to keep the flexibility and minimal knowledge about the various types of detectives at the precinct layer.

Reply:
So the tool I'm using is `madge`. I did test that it's possible to pass these options from madge (even via `.madgerc`) but I considered it appropriate to handle at this layer because:

1. This is the layer that selects behaviour based on file extensions. 
2. This matches the behaviour of the typescript compiler (and other tools): JSX can only be parsed in tsx files, and a tsx file extension always enables JSX parsing. Put another way: you always want this option enabled for tsx files. 

However, I can see that having this special case `if` right here isn't particularly elegant. What would you think about providing an alternate `tsx` export from detective-typescript and selecting that in the switch statement?

Reply:
Thanks for the additional context. What about enabling the flag by default in detective-typescript?

Reply:
On by default would break parsing of the older style type assertions: `const x = <SomeType> foo;` (asserts that foo is of type `SomeType`). I'm pretty sure this is the reason the Typescript compiler mandates separate file extensions.

Reply:
On by default would break parsing of the older style type assertions: `const x = <SomeType> foo;` (asserts that foo is of type `SomeType`). I'm pretty sure this is the reason the Typescript compiler mandates separate file extensions.

Reply:
On by default would break parsing of the older style type assertions: `const x = <SomeType> foo;` (asserts that foo is of type `SomeType`). I'm pretty sure this is the reason the Typescript compiler mandates separate file extensions.

Reply:
Ah, thanks for clarifying.

> What would you think about providing an alternate tsx export from detective-typescript and selecting that in the switch statement?

That works well for keeping the config closest to the lib that maintains it and keeping precinct as dumb as possible.

Thanks for your effort here. It's always great seeing community contributions ❤️